include/dlaf/matrix_output.h

```diff
@@ -0,0 +1,46 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+/// @file
+
+#include "dlaf/matrix.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf {
+
+template <class T>
+void printElements(Matrix<T, Device::CPU>& mat) {
```

Comment:
I'm not really sure this is a needed function. If you all think we want this, I would propose to at least put it in the `matrix` namespace (and maybe the name can be improved, or better I would go for making it `std::ostream& operator<<`).

IMHO, it is something we don't want to provide in the library, even if it can be useful during debugging (in case I would move it DLAF_test library).

Reply:
ostream operator is already used for matrix info (shape, ...)

Reply:
Anyway, this is a utility you want to have in the library, since you don't want to link DLAF_test to the main application.

Reply:
Ok, I propose to move this work about printing tiles and matrices to a dedicated PR. It is already a quite big PR and there is no need of it here. Do you agree?